Don’t require full vertex for expand

kmcginnes opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/aws/graph-explorer/blob/main/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/aws/graph-explorer/blob/main/CONTRIBUTING.md) before opening a pull request.
-->

## Description

Simplifies the expand node api just a bit by removing the need to send a full `Vertex` object.

Expanding a node only requires the vertex ID. Except for SPARQL. It requires the `vertexType` as well, which is used to path `Edge` instances during mapping with the source vertex type.

In the future I will remove the need to send the `vertexType` entirely when I redo all SPARQL queries. For now it remains

## Validation

- Tested on all three query engines

## Related Issues

- Tangentially related to #808

### Check List

<!--
  ATTENTION
  Please follow this check list to ensure that you've followed all items before opening this PR
  You can check the items by adding an `x` between the brackets, like this: `[x]`
-->

- [x] I confirm that my contribution is made under the terms of the Apache 2.0
      license.
- [x] I have run `pnpm checks` to ensure code compiles and meets standards.
- [x] I have run `pnpm test` to check if all tests are passing.
- [ ] I have covered new added functionality with unit tests if necessary.
- [ ] I have added an entry in the `Changelog.md`.

kmcginnes commented on packages/graph-explorer/src/hooks/useExpandNode.tsx:
```diff
@@ -56,10 +57,8 @@ export default function useExpandNode() {
-          expandNodeRequest.vertex.types?.join("::") ??
-          expandNodeRequest.vertex.type,
+        vertexId: expandNodeRequest.vertexId,
+        vertexType: expandNodeRequest.vertexType,
```
I moved the type joining logic to the two places that called this function. However, in reality it is not necessary since SPARQL vertices only ever have one type.

andreachild replied:
Can the type joining be removed then if it's not necessary?

acarbonetto replied:
if that's the case, could you not just hardcode or default the type for SPARQL?

Cole-Greer replied:
Is the vertexType just the label(s) for property graphs or the rdf:type for SPARQL?

kmcginnes replied:
@Cole-Greer `vertexType` is the node label for property graphs and the resource class or `rdf:type` for SPARQL, yes.

acarbonetto replied:
optional then?

kmcginnes replied:
@acarbonetto These vertex types are runtime dependent. For SPARQL we are simply sending the vertex type that was retrieved previously (typically during search) to the expansion logic so that it can be applied to all the `Edge` instances being returned. Those edge instances have the source and target node ID and the source and target types (labels/class). But since RDF doesn't really have "edges" we have to fake it. This is part of that fakery.

acarbonetto replied:
I'm just thinking that if you want to remove these entirely in the future, you should consider making the variable optional `vertexType?` and that will make it easier in the future to remove it.

kmcginnes replied:
The future changes I refer to would be to change the SPARQL query to query for the source node class so that it doesn't have to be passed in to the request.

I don't want to make that change right now because doing so would require breaking the support for blank nodes. I have approval to remove support for blank nodes, but I don't want to break that support until I am ready to make more improvements to all the SPARQL queries.

kmcginnes replied:
It is optional for gremlin and openCypher, but required for SPARQL. Most of the logic in Graph Explorer is agnostic to the query engine selected.

So I could throw an error to validate SPARQL requests I guess. But I like the type safety of requiring the parameter be provided, even if it is not used by 2 out of 3 code paths.

andreachild commented:
> In the future I will remove the need to send the vertexType entirely when I redo all SPARQL queries. For now it remains

Nit: would be nice to add a comment that the type is intended to be removed in the future because otherwise it seems unnecessary to replace Vertex with id + type since it already contains the information that is used downstream.

acarbonetto commented on packages/graph-explorer/src/connector/useGEFetchTypes.ts:
```diff
@@ -82,6 +82,8 @@ export type NeighborsRequest = {
+   *
+   * Used only by the SPARQL implementation to patch the edges returned with the source vertex type.
    */
   vertexType: string;
```
optional?

kmcginnes replied:
Unfortunately, it is required by 1 out of the 3 code paths. Throwing an exception would make for a worse developer experience, in my opinion, since it moves the failure to runtime (and only if you test that code path).

kmcginnes commented:
To hopefully clean up the code a bit more I made a few additional changes:

* Change `Vertex.types` to be required
  * This forced a change in the SPARQL vertex mapping code to add `types` array
  * This broke some tests that I fixed
* Change the expand API to require the types array instead of a single string representing the types
  * I moved the logic joining the types array to a single string in to the sparql code that does the mapping
  * Fixed the broken tests
=
